### DISCUSSION TITLE
packageRule to only match in-range updates

### DISCUSSION THREAD

**Discussion by jacobrask** [renovatebot/renovate#14858] [Request Help]
In our monorepo, I want any updates that can be done in-range without updating the package.json of an application to be done in a single PR shared for all apps. If the change is not in-range, it should be a separate PR for that application.

I have this configuration to create app specific PRs

```
    {
      "matchPaths": ["apps/**"],
      "matchDepTypes": ["dependencies"],
      "additionalBranchPrefix": "{{parentDir}}-",
      "commitMessagePrefix": "{{parentDir}} |"
    },
```

I tried adding this rule after the one above

```
    {
      "matchUpdateTypes": ["digest"],
      "additionalBranchPrefix": "",
      "commitMessagePrefix": ""
    },
```

but Renovate is still creating application level PRs for patch updates to dependencies with a `^` range. Am I misunderstanding what `digest` means? Is there any other way to accomplish this?

**Comment by viceice** [comment]
https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

**Reply by jacobrask** [reply]
That's not what I'm looking for though - I want to match the updates where "package `foo` will be updated from 1.0.0 to 1.0.1 and package.json specifies `"foo": "^1.0.0"` - i.e. "in-range" changes that require changes to the lockfile but not to package.json.

> **update-lockfile** = Update the lock file when in-range updates are available, otherwise replace for updates out of range. Works for bundler, composer, npm, yarn, terraform and poetry so far
> **in-range-only** = Update the lock file when in-range updates are available, ignore package file updates

https://docs.renovatebot.com/configuration-options/#rangestrategy

**Reply by jacobrask** [reply]
The intention is to let the owner of a package.json file set different levels of trust in a dependency using npm ranges the way they are intended - if you trust a package to not do breaking changes in minor versions, you use a `^` range, if you trust it to not do changes in patches you use a `~` range, otherwise you pin the dependency and get a separate PR for any update.

**Comment by rarkins** [comment]
I don't think we have any metadata currently for matching in-range only updates

**Reply by jacobrask** [reply]
Ok I see. So is `digest` a Docker specific thing, or what does it mean? Can't find it described in the docs.

I'll try if `"additionalBranchPrefix": "{{#unless (containsString newValue currentValue)}}{{parentDir}}-"{{/unless}}"` works - to create a separate PR only if currentValue and newValue are different 🤔

**Reply by rarkins** [reply]
Digest is mostly used for Docker, but not exclusively. For example if a git branch reference is pinned to a SHA then the SHA can update, and this is also known as a digest update.

**Comment by mvanduijker** [comment]
I see this question open for quite a bit, is there a solution / workaround available?
